ddtrace/writer.py

```diff
@@ -83,21 +83,23 @@ def flush_queue(self):
             traces = self._apply_filters(traces)
         except Exception as err:
             log.error('error while filtering traces: {0}'.format(err))
+            return
 
-        traces_response = None
+        traces_responses = None
 
         if traces:
             # If we have data, let's try to send it.
-            try:
-                traces_response = self.api.send_traces(traces)
-            except Exception as err:
-                log.error('cannot send spans to {1}:{2}: {0}'.format(
-                    err, self.api.hostname, self.api.port))
-
-        if self._priority_sampler and traces_response:
-            result_traces_json = traces_response.get_json()
-            if result_traces_json and 'rate_by_service' in result_traces_json:
-                self._priority_sampler.set_sample_rate_by_service(result_traces_json['rate_by_service'])
+            traces_responses = self.api.send_traces(traces)
+            for response in traces_responses:
+                if isinstance(response, Exception):
+                    log.error('cannot send spans to {1}:{2}: {0}'.format(
+                        response, self.api.hostname, self.api.port))
+
+        if self._priority_sampler and traces_responses:
```

Comment:
Should we roll this into the for loop above?

Saves on iterating over the list twice.

Also, do we need to check if `traces_response` is an exception first before we call `.get_json()`?

Reply:
(maybe this shows a test case we are missing)

Reply:
Yes we can. I'm really trying to limit the scope of my patch each time, so I tend to avoid doing that, but if you think it's worth it, I can make my PR longer :)

Reply:
(refactoring solves the exception check so 👍 )